### PR TITLE
ente-cli: init at 0.2.2

### PR DESCRIPTION
**Pull request by zi3m5f**
## Description of changes

Part of: #292641

Add cli client for ente.io server, see #325659.

Export of photos works for me with official server.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/en/ente-cli/package.nix [hunk 1]
```diff
@@ -0,0 +1,69 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+  nix-update-script,
+  ...
```
Comment (dotlambda) [inline pkgs/by-name/en/ente-cli/package.nix:6]:
```suggestion
```

File: pkgs/by-name/en/ente-cli/package.nix [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    hash = "sha256-gIDJUj2pn8rndXWN69bZdVfRLVB7AybXHMcioG2NI1k=";
+    fetchSubmodules = false;
+    sparseCheckout = [ "cli" ];
+  };
+
+  modRoot = "./cli";
+
+  vendorHash = "sha256-Gg1mifMVt6Ma8yQ/t0R5nf6NXbzLZBpuZrYsW48p0mw=";
+
+  CGO_ENABLED = 0;
```
Comment (dotlambda) [inline pkgs/by-name/en/ente-cli/package.nix:29]:
```suggestion
  env.CGO_ENABLED = 0;
```

File: pkgs/by-name/en/ente-cli/package.nix [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X main.AppVersion=cli-v${version}"
+  ];
+
+  postInstall = ''
+    mv $out/bin/{cli,${pname}}
```
Comment (dotlambda) [inline pkgs/by-name/en/ente-cli/package.nix:38]:
```suggestion
    mv $out/bin/{cli,ente-cli}
```
#277994

File: pkgs/by-name/en/ente-cli/package.nix [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    platforms = [
+      "aarch64-linux"
+      "armv7a-linux"
+      "i686-linux"
+      "x86_64-linux"
+      "x86_64-darwin"
+      "aarch64-darwin"
+      "i686-windows"
+      "x86_64-windows"
+    ];
```
Comment (dotlambda) [inline pkgs/by-name/en/ente-cli/package.nix:66]:
```suggestion
```

File: pkgs/by-name/en/ente-cli/package.nix [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+  passthru.updateScript = nix-update-script {
+    extraArgs = [
+      "--version-regex"
+      "cli-(.+)"
+    ];
+  };
+
+  meta = {
+    description = "CLI client for downloading your data from Ente.";
```
Comment (dotlambda) [inline pkgs/by-name/en/ente-cli/package.nix:49]:
```suggestion
    description = "CLI client for downloading your data from Ente";
```
https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#meta-attributes

File: pkgs/by-name/en/ente-cli/package.nix [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+  nix-update-script,
+  ...
+}:
+let
+  pname = "ente-cli";
+  version = "0.2.1";
+
```
Comment (dotlambda) [inline pkgs/by-name/en/ente-cli/package.nix:11]:
```suggestion
```

File: pkgs/by-name/en/ente-cli/package.nix (hunk 4 above)
Comment (dotlambda) [reply pkgs/by-name/en/ente-cli/package.nix:64]: In pretty sure `buildGoModule` sets it automatically.

Comment (dotlambda) [reply pkgs/by-name/en/ente-cli/package.nix:64]: > Should we really set/support more platforms than [upstream](https://github.com/ente-io/ente/blob/4f74470abe8d36dc203fe85d399b0a63d01e85c8/.github/workflows/cli-release.yml#L31)?

Yes. If you want to make sure they actually work you should make the derivation run unittests.

File: pkgs/by-name/en/ente-cli/package.nix [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+  CGO_ENABLED = 0;
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X main.AppVersion=cli-v${version}"
+  ];
+
+  postInstall = ''
+    mv $out/bin/{cli,ente-cli}
```
Comment (teto) [inline pkgs/by-name/en/ente-cli/package.nix:37]: would be cool to add completion as well, the --help shows a `completion` command

**Comment by teto** [comment]
I've never used ente but I am curious to try. Running this executable:
```
➜ ente-cli 
Start by creating a config file in your home directory:

Usage:
  ente [flags]
  ente [command]
```
it seems that the executable is supposed to be called "ente" ?

File: pkgs/by-name/en/ente-cli/fix-env-sh.nix [hunk 8]
```diff
@@ -0,0 +1,16 @@
+{ lib, dbus }:
+
+''
+  # adding dbus to path fixes:
+  #> ente-cli-test> error getting password from keyring: exec: "dbus-launch": executable file not found in $PATH
+  export PATH="$PATH:${lib.makeBinPath [ dbus ]}"
```
Comment (dotlambda) [inline pkgs/by-name/en/ente-cli/fix-env-sh.nix:6]: Why not put `dbus` in `nativeBuildInputs`?

File: pkgs/by-name/en/ente-cli/test-simple.nix [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+  ente-cli,
+  runCommand,
+}:
+
+runCommand "ente-cli-test" { } ''
+  set -euo pipefail
+
+  ${callPackage ./fix-env-sh.nix { }}
+
+  ${lib.getExe ente-cli} version > $out
```
Comment (dotlambda) [inline pkgs/by-name/en/ente-cli/test-simple.nix:13]: Can we use `versionCheckHook` instead?

**Review by FliegendeWurst — APPROVED** [review]

**Review by Aleksanaa — APPROVED** [review]

**Comment by Aleksanaa** [comment]
@ofborg build ente-cli

**Comment by Aleksanaa** [comment]
It works with `nixpkgs-review` but not ofborg, having no idea

**Comment by Aleksanaa** [comment]
See https://github.com/NixOS/nixpkgs/issues/308283. I'm not strongly opposed to it but as the below comment mentioned, adding an emulator is an overhead.

**Pull request merged**